### PR TITLE
Feature/submissions and assessments/assessing before due date warning

### PR DESCRIPTION
**Pull request by HanyaElhashemy**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] ~~Client: I added multiple integration tests (Jest) related to the features~~
- [x] ~~Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~~
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Description

Currently, instructors have full access rights to all operations of Artemis, including assessing a submission although the due date is not over yet. In some situations, this may lead to students sharing sample solutions with other students before the due date. Therefore we should warn the instructors in the "submissions and assessments" page if the due date is not over yet.

### Steps for Testing

Make sure to login as an instructor
1. Navigate to Overview and click on the exercise detail page of Text|Modeling|FileUpload exercises
2. Click on "submissions" under instructor actions
3. A warning sign should be displayed 
4. Navigate to Tutor Exercise Dashboard
A warning sign should be displayed there as well

### Screenshots

image

image

### CODE REVIEW

File: src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<fa-icon
```
Comment (lovaiible) [inline src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.html]: Just personal preference: you can add the html directly in the `assessment-warning.component.ts` since it's just 8 lines of code.

File: src/main/webapp/i18n/en/assessment.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
                 "updateAfterComplaintFailed": "Updating the assessment failed."
             },
             "dashboard": {
+                "warning": "Note, the due date has not passed yet",
```
Comment (lovaiible) [inline src/main/webapp/i18n/en/assessment.json]: Maybe "Attention! ..." is a better wording than "Note, .." ?

Comment (lovaiible) [inline src/main/webapp/i18n/en/assessment.json]: Maybe "Attention! ..." is a better wording than "Note, .." ?

File: src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.html (hunk 1 above)
Comment (lovaiible) [inline src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.html]: Just personal preference: you can add the html directly in the `assessment-warning.component.ts` since it's just 8 lines of code.

**Comment by maxr96** [comment]
Good work @HanyaElhashemy. I agree with comments from @lovaiible. And I am also wondering if this should be even possible for instructors to assess before the due date. Are there any specific situations when we need to do so?
Although if we still need this functionality, maybe we can add a confirmation window as well? Since this should be a rare case, I guess it is a good idea to warn instructor better.

File: src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+/*
+Warn instructors in the assessment and submission page
+if the due date is not over yet as they are allowed to assess submissions there
+ */
+@Component({
+    selector: 'jhi-assessment-warning',
+    templateUrl: './assessment-warning.component.html',
+})
+export class AssessmentWarningComponent implements OnChanges {
+    @Input()
```
Comment (maxr96) [inline src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.ts]: Can you please move it to the line where the argument is defined?

File: src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    exercise: Exercise;
+    currentDate: moment.MomentInput;
+    isBeforeDueDate = false;
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const dueDate = this.exercise.dueDate;
```
Comment (maxr96) [inline src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.ts]: So I guess this should be only triggered when `exercise` changes. Can we move it to the `exercise` setter?

File: src/main/webapp/app/file-upload-assessment/file-upload-assessment-dashboard/file-upload-assessment-dashboard.component.ts [hunk 5]
```diff
@@ -28,7 +28,6 @@ export class FileUploadAssessmentDashboardComponent implements OnInit {
     reverse = false;
 
     private cancelConfirmationText: string;
-
```
Comment (maxr96) [inline src/main/webapp/app/file-upload-assessment/file-upload-assessment-dashboard/file-upload-assessment-dashboard.component.ts]: Please revert this change

File: src/main/webapp/i18n/en/assessment.json [hunk 6]
```diff
@@ -29,6 +29,7 @@
                 "updateAfterComplaintFailed": "Updating the assessment failed."
             },
             "dashboard": {
+                "warning": "Warning: the due date is not over yet and students are still allowed to submit their solutions.Please do not start with the assessment yet!",
```
Comment (lovaiible) [inline src/main/webapp/i18n/en/assessment.json]:
```suggestion
                "warning": "Warning: the due date is not over yet and students are still allowed to submit their solutions. Please do not start with the assessment yet!",
```

File: src/main/webapp/i18n/de/assessment.json [hunk 7]
```diff
@@ -29,6 +29,7 @@
                 "updateAfterComplaintFailed": "Das Aktualisieren der Bewertung ist fehlgeschlagen."
             },
             "dashboard": {
+                "warning": "Achtung: der Abgabetermin ist noch nicht abgelaufen und Studenten dürfen noch ihre Einreichungen abgeben.Bitte noch nicht mit der Korrektur anzufangen",
```
Comment (lovaiible) [inline src/main/webapp/i18n/de/assessment.json]:
```suggestion
                "warning": "Achtung: der Abgabetermin ist noch nicht abgelaufen und Studenten dürfen noch ihre Einreichungen abgeben. Bitte noch nicht mit der Korrektur anfangen",
```

**Comment by HanyaElhashemy** [comment]
> Looks good to me :)

Thank you for reviewing my PR and for helping with fixing the tests and running prettier ^^
